core/controllers/acl_decorators.py

```
@@ -5133,6 +5133,71 @@ def test_can_access_certificate_dashboard(
     return test_can_access_certificate_dashboard
 
 
+def can_submit_assessment_response(
+    handler: Callable[..., _GenericHandlerFunctionReturnType],
+) -> Callable[..., _GenericHandlerFunctionReturnType]:
+    """Checks whether the current learner can submit the attempt."""
+
+    @functools.wraps(handler)
+    def test_can_submit(
+        self: _SelfBaseHandlerType, attempt_id: str, **kwargs: Any
```

Ruff (0.16.0)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)